### PR TITLE
Abehern/vz 7560 enforce ut branch coverage gt or eq master

### PR DESCRIPTION
**Pull request by AbeHern**
Created new boolean parameter in Jenkinsfile to fail build if master or release if UT coverage % is less than previous master or release.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -262,6 +266,13 @@ pipeline {
                     cd ${GO_REPO_PATH}/verrazzano
                     make precommit
                 """
+                        script{
```
Comment (kminder) [inline Jenkinsfile:269]: This is the type of thing we need to figure out how to do from the Makefile vs the Jenkinsfile.

File: platform-operator/internal/vzconfig/args_test.go [hunk 2]
```diff
@@ -87,14 +86,14 @@ func TestCheckExternalIPsOverridesArgsWithPaths(t *testing.T) {
...
 	override := fmt.Sprintf(`spec:
-  components:
-    ingressGateways:
-      - name: istio-ingressgateway
-        k8s:
-          service:
-            type: NodePort
-            externalIPs:
-            - %s`, externalIP)
+ components:
```
Comment (kminder) [inline platform-operator/internal/vzconfig/args_test.go:89]: Is this intentional?

File: build/coverage-number-comparison.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Copyright (c) 2020, 2022, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+#Get latest line-rate from master/release for comparison
+LOCAL_BRANCH_NAME=$(git branch  --no-color  | grep -E '^\*' | sed 's/\*[^a-z]*//g')
+OBJECT_URL=https://objectstorage.us-phoenix-1.oraclecloud.com/n/stevengreenberginc/b/verrazzano-builds/o/master/unit-test-coverage-number.txt
```
Comment (agv426) [inline build/coverage-number-comparison.sh:9]: This should be constructed from env variables (which may be coming from creds on Jenkins)

File: Jenkinsfile [hunk 4]
```diff
@@ -39,6 +39,7 @@ pipeline {
         booleanParam (description: 'Whether to fail the Integration Tests to test failure handling', name: 'SIMULATE_FAILURE', defaultValue: false)
         booleanParam (description: 'Whether to perform a scan of the built images', name: 'PERFORM_SCAN', defaultValue: false)
         booleanParam (description: 'Whether to wait for triggered tests or not. This defaults to false, this setting is useful for things like release automation that require everything to complete successfully', name: 'WAIT_FOR_TRIGGERED', defaultValue: false)
+        booleanParam (description: 'Whether to fail a build or pass based on its UT coverage %. This will only issue a warning when not building master or release.', name: 'COVERAGE_TEST', defaultValue: false)
```
Comment (jmaron99) [inline Jenkinsfile:42]: perhaps a name such as "COVERAGE_GATES_BUILD"?

**Pull request closed without merging**